src/platform/macos/cgl/connection.rs

```diff
@@ -125,15 +125,27 @@ impl Connection {
         raw_handle: rwh_05::RawWindowHandle,
         _size: Size2D<i32>,
     ) -> Result<NativeWidget, Error> {
-        use crate::platform::macos::system::surface::NSView;
-        use cocoa::base::id;
+        use objc2::{rc::Retained, MainThreadMarker};
+        use objc2_app_kit::{NSView, NSWindow};
         use rwh_05::RawWindowHandle::AppKit;
 
         match raw_handle {
-            AppKit(handle) => Ok(NativeWidget {
-                view: NSView(unsafe { msg_send![handle.ns_view as id, retain] }),
-                opaque: unsafe { msg_send![handle.ns_window as id, isOpaque] },
-            }),
+            AppKit(handle) => {
+                assert!(
+                    MainThreadMarker::new().is_some(),
+                    "NSView is only usable on the main thread"
+                );
+                // SAFETY: The pointer is valid for as long as the handle is,
+                // and we just checked that we're on the main thread.
+                let ns_view = unsafe { Retained::retain(handle.ns_view.cast::<NSView>()).unwrap() };
+                let ns_window =
+                    unsafe { Retained::retain(handle.ns_window.cast::<NSWindow>()).unwrap() };
```

Comment:
Is it important to explicitly retain the window here if you are simply calling `isOpaque()` and discarding the smart pointer right after?

Reply:
Nope, it isn't, I've changed it